components/suggestion/generic_channel_provider.tsx

```diff
@@ -53,24 +65,25 @@ class ChannelSuggestion extends Suggestion {
 }
 
 export default class ChannelProvider extends Provider {
-    constructor(channelSearchFunc) {
+    autocompleteChannels: any;
+    constructor(channelSearchFunc: (term: string, success: (channels: Channel[]) => void, error: (err: ServerError) => void) => ActionResult<any, any> | Promise<ActionResult<any, any> | Array<ActionResult<any, any>>>) {
```

Comment:
Ditto on using `ActionFunc`. For this file, we can create a type alias for reuse, something like:

```
type ChannelSearchFunc = (term: string, success: (channels: Channel[]) => void, error: (err: ServerError) => void) => ActionFunc

export default class ChannelProvider extends Provider {
    autocompleteChannels: ChannelSearchFunc
    constructor(channelSearchFunc: ChannelSearchFunc) {
        super();

        this.autocompleteChannels = channelSearchFunc;
    }

```